flamegraph/flamegraph.py

```diff
@@ -109,6 +110,16 @@ def start_profile_thread(fd, interval=0.001, filter=None, format_entry=default_f
   profile_thread.start()
   return profile_thread
 
+@contextlib.contextmanager
+def profile(*args, **kwds):
+  """Profile active threads in the background."""
```

Comment:
**Note to reviewers**: this context manager is extacted from code in `main()`.  It makes it easier to unit test the profiling thread without executing an external script.